lib/arborist/reify.js

```diff
@@ -887,6 +886,18 @@ module.exports = cls => class Reifier extends cls {
       filter: diff => diff.action === 'ADD' || diff.action === 'CHANGE',
     })
 
+    // pick up link nodes from the unchanged list as we want to run their
+    // scripts in every install despite of having a diff status change
+    for (const node of this.diff.unchanged) {
+      const tree = node.root.target || node.root
+
+      // skip links that only live within node_modules as they are most
+      // likely managed by packages we installed, we only want to rebuild
+      // unchanged links we directly manage
+      if (node.isLink && node.target.fsTop === tree)
```

Comment:
Could also consider `|| !this.diff.unchanged.includes(node.target.fsTop)` to capture the "did not change link, but did change link's fsParent" use case.  (Separate issue from the one this commit addresses, can come later.)